packages/cli/src/cli/config/config.update.ts

```diff
@@ -56,15 +57,20 @@ export class Updater<S extends BaseConfig = BaseConfig> {
     else return `/v1/tiles/${this.config.id.slice(3)}/*`;
   }
 
+  async getOldData(): Promise<ConfigImagery | ConfigTileSet | ConfigProvider | ConfigVectorStyle | null> {
+    if (this.oldData) return this.oldData;
```

Comment:
this will trigger a fetch multiple times if `oldData` is null.

your best to store oldData fetch as a `Promise` 

```typescript
getOldData(): Promise<...> { // no async
  if (this.oldData) return this.oldData;
  this.oldData = this.getDb().get(this.config.id)` // (no await)
  return this.oldData
}
```

then you can just `const oldData = await this.getOldData()`

Reply:
Fixed. Thank you.